<?xml version="1.0"?>
<launch>
  <!-- Frame fijo de RViz y de los markers -->
  <arg name="fixed_frame" default="odom"/>
  <arg name="rviz" default="true"/>

  <!-- Visualizador -->
  <node pkg="racecar_track_geometry" type="track_viz.py" name="track_viz" output="screen">
    <rosparam command="load" file="$(find racecar_track_geometry)/config/track_viz.yaml"/>
    <!-- Si quieres forzar frame desde launch:
    <param name="frame_id" value="$(arg fixed_frame)"/>
    -->
    <!-- Publica: ~markers (MarkerArray), ~centerline_path (Path) -->
  </node>

  <!-- (Opcional) Lanza RViz con perfil -->
  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find racecar_track_geometry)/rviz/track_debug.rviz"/>
  </group>
</launch>
